uberfire-structure/uberfire-structure-backend/src/main/java/org/guvnor/structure/backend/pom/DependencyTypesMapper.java

```diff
@@ -26,18 +28,30 @@
 public class DependencyTypesMapper {
 
     private final static String JSON_POM_DEPS = "DependencyTypesMapper.json";
-    private Map<DependencyType, List<DynamicPomDependency>> mapping;
+    private ConfigurationMap conf;
 
     public DependencyTypesMapper() {
-        PomJsonReaderDefault jsonDepsReader = new PomJsonReaderDefault(getClass().getClassLoader().getResourceAsStream(JSON_POM_DEPS));
-        mapping = jsonDepsReader.readDeps();
+        PomJsonReader jsonDepsReader = new PomJsonReaderDefault(getClass().getClassLoader().getResourceAsStream(JSON_POM_DEPS));
+        conf = jsonDepsReader.readConfiguration();
+    }
+
+    public String getKieVersion() {
+        return conf.getKieVersion();
     }
 
     public Map<DependencyType, List<DynamicPomDependency>> getMapping() {
-        return mapping;
+        return conf.getMapping();
+    }
+
+    public Set<DynamicPomDependency> getInternalArtifacts() {
+        return conf.getInternalArtifacts();
     }
 
-    public List<DynamicPomDependency> getDependencies(DependencyType key) {
-        return mapping.get(key);
+    public List<DynamicPomDependency> getDependencies(Set<DependencyType> dependencyTypes) {
+        List<DynamicPomDependency> result = new ArrayList<>();
+        for (DependencyType type : dependencyTypes) {
+            result.addAll(conf.getMapping().get(type));
```

Comment:
It's not better and safe check if there is a configuration for that type on the JSON?

Reply:
Added a check about the presence